tests/bot_wallet_test.py

```diff
@@ -0,0 +1,303 @@
+"""System test for checking calculated wallets versus wallets on chain"""
+from __future__ import annotations
+
+import logging
+import os
+from dataclasses import dataclass
+from typing import cast
+
+from agent0 import build_account_key_config_from_agent_config
+from agent0.base import MarketType, Trade
+from agent0.base.config import AgentConfig, EnvironmentConfig
+from agent0.hyperdrive.exec import run_agents
+from agent0.hyperdrive.policies import HyperdrivePolicy
+from agent0.hyperdrive.state import HyperdriveActionType, HyperdriveMarketAction, HyperdriveWallet
+from eth_typing import URI
+from ethpy import EthConfig
+from ethpy.base import smart_contract_read
+from ethpy.hyperdrive import AssetIdPrefix, encode_asset_id
+from ethpy.hyperdrive.addresses import HyperdriveAddresses
+from ethpy.hyperdrive.api import HyperdriveInterface
+from ethpy.test_fixtures.local_chain import DeployedHyperdrivePool
+from fixedpointmath import FixedPoint
+from numpy.random._generator import Generator as NumpyGenerator
+from web3 import HTTPProvider
+
+
+def ensure_agent_wallet_is_correct(wallet: HyperdriveWallet, hyperdrive: HyperdriveInterface) -> None:
+    """Function to check that the agent's wallet matches what's reported from the chain.
+    Will assert that balances match
+
+    Arguments
+    ---------
+    wallet: HyperdriveWallet
+        The HyperdriveWallet object to check against the chain
+    hyperdrive: HyperdriveInterface
+        The Hyperdrive API interface object
+    """
+    # Check base
+    base_from_chain = smart_contract_read(hyperdrive.base_token_contract, "balanceOf", wallet.address)["value"]
```

Comment:
FYI there is a [hyperdrive interface function](https://github.com/delvtech/elf-simulations/blob/main/lib/ethpy/ethpy/hyperdrive/api/api.py#L338) for this.

Reply:
Although looking at the rest of the code it probably makes sense to leave it as a low-level call

Reply:
Right, I think I considered using this, but I didn't have access here to the agent object.